static/js/compose.js

```diff
@@ -538,6 +544,13 @@ exports.transmit_message = function (request, success, error) {
         send_message_ajax(request, success, error);
     }
 };
+function report_content_not_sent(request) {
```

Comment:
Yeah, I think you'll want two separate tests here.  Your first test looks great, and the second test would 
have similar structure, but just never simulate successful message transmission.

I don't think you need to bend over backward to remove duplicated code between the tests, but if you can at least make a couple helper functions to reduce some duplication, that would be helpful.